lib/msf/core/db_manager/import/qualys/asset.rb

```diff
@@ -54,28 +57,32 @@ def import_qualys_asset_xml(args={}, &block)
     data = args[:data]
     wspace = Msf::Util::DBManager.process_opts_workspace(args, framework).name
     bl = validate_ips(args[:blacklist]) ? args[:blacklist].split : []
-    doc = rexmlify(data)
+    doc = Nokogiri.XML(data)
     vuln_refs = find_qualys_asset_vuln_refs(doc)
 
     # 2nd pass, actually grab the hosts.
-    doc.elements.each("/ASSET_DATA_REPORT/HOST_LIST/HOST") do |host|
+    doc.xpath("/ASSET_DATA_REPORT/HOST_LIST/HOST").each do |host|
       hobj = nil
-      addr = host.elements["IP"].text if host.elements["IP"]
+      addr_el = host.xpath("IP").first
+      addr = addr_el.text if addr_el
       next unless validate_ips(addr)
```

Comment:
This will cause addr to be undefined if `addr_el` is not defined. Subsequently the undefined `addr` value will then be used in a `validate_ips` call. Might want to fix this.

Reply:
The same was true before is `host.elements["IP"]` was missing `addr` would have been `nil`, and passed to `validate_ips`.

Reply:
Which will cause a crash as https://github.com/rapid7/metasploit-framework/blob/04e8752b9b74cbaad7cb0ea6129c90e3172580a2/lib/msf/core/db_manager/ip_address.rb#L58-L61 shows that we will not check if the parameter is `nil` and will instead just call `split` on the parameter.

Likely the best fix would be to actually update `validate_ips` itself but I just wanted to point out this is still an outstanding issue and goes back to a discussion I was having with Spencer about having libraries do more validation in the methods themselves vs relying on callers doing their own validation.

If we want to push this into a separate PR then fine, but I don't think ignoring this is the solution here.

Reply:
Edit: Thought this would cause an issue and most of the time it would however it seems this issue was predicted somehow cause the code uses a begin...rescue clause that will take care of this edge case, so we are all good